stylelint.config.js

```diff
@@ -1,27 +1,49 @@
 // eslint-disable-next-line @typescript-eslint/no-var-requires
 const path = require('path')
 
+const BEM_REGEX = new RegExp(
+	/^[a-z]([-]?[a-z0-9]+)*(__[a-z0-9]([-]?[a-z0-9]+)*)?(--[a-z0-9]([-]?[a-z0-9]+)*)?$/,
+)
+
 module.exports = {
 	configBaseDir: path.resolve(__dirname, 'node_modules'),
-	plugins: ['stylelint-csstree-validator', 'stylelint-selector-no-empty'],
+	plugins: ['stylelint-csstree-validator'],
 	extends: [
-		'stylelint-config-recommended-scss',
+		'stylelint-config-standard',
+		'stylelint-config-standard-scss',
 		'stylelint-config-recess-order',
 		'stylelint-prettier/recommended',
-		'stylelint-scss',
+		'stylelint-config-recommended-vue',
 	],
 	rules: {
+		'at-rule-no-unknown': null,
+		'function-no-unknown': null,
 		'no-descending-specificity': null,
 		'no-empty-source': null,
-		'scss/selector-no-redundant-nesting-selector': null,
-		'scss/no-duplicate-dollar-variables': true,
-		'csstree/validator': true,
-		'plugin/stylelint-selector-no-empty': true,
+		'selector-class-pattern': [
+			BEM_REGEX,
+			{
+				resolveNestedSelectors: true,
+				message: (error) =>
+					`Follow the [bem](https://getbem.com/naming/) convention for class naming, ${error} doesn't comply`,
+			},
+		],
 		'selector-pseudo-element-no-unknown': [
 			true,
 			{
 				ignorePseudoElements: ['v-deep'],
 			},
 		],
+		'scss/at-import-no-partial-leading-underscore': null,
+		'scss/at-import-partial-extension': null,
+		'scss/at-rule-no-unknown': true,
+		'scss/function-no-unknown': true,
+		'scss/no-duplicate-dollar-variables': true,
+		'value-keyword-case': [
```

Comment:
this rule is very dangerous and had a few auto effects that were bad, explained and listed below

1) it destroyed our font-family value because we had some values quoted and some values unquoted.. those that were unquoted were auto-capitalized.. for webkit browsers if a value doesn't match `@font-faces` , it goes to the next, for Mozilla, thhe auto-capitalization didn't matter and it managed to find the font-family regardless which meant that between gecko and webkit the resolution was different 

2) it broke the svg properties (it tried to automatically lowercase it again), which is not an accepted value.. we had to therefore override the default for casing them to be camel-casing rather than lower case

--
As a rule of thumb just reread what the rule autofixes and check that it is a correct value indeed